### PR TITLE
DNA Lab support

### PR DESCRIPTION
**Pull request by horrible-little-slime**

### CODE REVIEW

**Comment by horrible-little-slime** [comment]
Most functions here have been tested as part of manny-hccs. Requesting review now

File: src/resources/2014/DNALab.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+ * Hybridize yourself with the current contents of your syringe, if possible.
+ * @returns Whether or not we succeeded
+ */
+export function hybridize(): boolean {
+  if (get("_dnaHybrid")) return false;
+  if (!installed()) return false;
+  const currentSyringe = get("dnaSyringe");
+  if (!currentSyringe) return false;
+  const tonicPotion = phylaTonics.get(currentSyringe);
+  if (!tonicPotion) return false; //this line should never trigger
```
Comment (phulin) [inline src/resources/2014/DNALab.ts:140]:
```suggestion
  const tonicPotion = getTonic(currentSyringe);
```

File: src/resources/2014/DNALab.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  const expectedEffect = getModifier("Effect", tonicPotion);
+  cliExecute("camp dnainject");
+  return isHybridized(expectedEffect);
+}
+
+/**
+ * Makes tonics with whatever phylum is currently in your syringe
+ * @param {number} [amount=1] the number of tonics to make
```
Comment (phulin) [inline src/resources/2014/DNALab.ts:148]: please document return value

File: src/resources/2014/DNALab.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+ */
+export function makeTonic(amount: 1 | 2 | 3 = 1): boolean {
+  if (!installed()) return false;
+  const currentSyringe = get("dnaSyringe");
+  if (!currentSyringe) return false;
+  const tonicPotion = phylaTonics.get(currentSyringe);
+  const amountToMake = clamp(amount, 0, tonicsLeft());
+  if (!tonicPotion) return false; //this line should never trigger
```
Comment (phulin) [inline src/resources/2014/DNALab.ts:156]: same suggestion as above

**Review by phulin — APPROVED** [review]

**Pull request merged**